statviz/src/App.tsx

```diff
@@ -1,17 +1,110 @@
-import { Route, Routes } from "react-router-dom";
-import Dashboard from "@boxtribute/shared-components/statviz/dashboard/Dashboard";
+import { gql, useQuery } from "@apollo/client";
+import { Alert, AlertIcon, Flex, Heading, Spinner } from "@chakra-ui/react";
+
+import BoxtributeLogo from "./BoxtributeLogo";
+import StockDataFilter from "@boxtribute/shared-components/statviz/components/visualizations/stock/StockDataFilter";
+import ErrorCard, {
+  predefinedErrors,
+} from "@boxtribute/shared-components/statviz/components/ErrorCard";
+
+const RESOLVE_LINK = gql(`
+  query resolveLink($code: String!) {
+    # TODO: Configure generated gql.tada for the public schema.
+    resolveLink(code: $code) {
+      __typename
+      ... on ResolvedLink {
+        view
+        urlParameters
+        baseName
+        organisationName
+        data {
+          # TODO: Refactor query once other views are implemented.
+          ... on StockOverviewData {
+            facts {
+              productName
+              categoryId
+              gender
+              boxesCount
+              itemsCount
+              sizeId
+              tagIds
+              boxState
+              locationId
+            }
+            dimensions {
+              category {
+                id
+                name
+              }
+              size {
+                id
+                name
+              }
+              tag {
+                id
+                name
+                color
+              }
+              location {
+                id
+                name
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+function matchErrorMessage(errorMsg: string) {
+  switch (true) {
+    case errorMsg.includes("Expired"):
+      return `The link has expired.`;
+    default:
+      return `An unexpected error happened: ${errorMsg}`;
+  }
+}
 
-// test precommit
 function App() {
+  const searchParams = new URLSearchParams(window.location.search);
+  const codeParam = searchParams.get("code");
+
+  if (codeParam) localStorage.setItem("code", codeParam);
+
+  const code = codeParam ?? localStorage.getItem("code");
+
+  const { data, loading, error } = useQuery(RESOLVE_LINK, { variables: { code } });
+
+  if (error) {
+    return (
+      <Alert status="error">
+        <AlertIcon />
+        {matchErrorMessage(error.message)}
+      </Alert>
+    );
+  }
+
+  if (loading) return <Spinner />;
+
+  if (data === undefined) return <ErrorCard error={predefinedErrors.noData} />;
+
+  if (codeParam) {
+    const newParams = `view=${data?.resolveLink?.view.toLowerCase()}&${data?.resolveLink?.urlParameters}`;
```

Comment:
would be nice to have handling of ExpiredLinkError or UnknownLinkError, otherwise this line fails.

Reply:
also urlParameters can be `null` which should be handled